src/Units/NonArmyWrappers/WorkerWrapper.cpp

```diff
@@ -1,24 +1,60 @@
 #include "./WorkerWrapper.h"
+#include "../../Strategist/Strategist.h"
+#include "BWEB.h"
+
+void WorkerWrapper::onFrame()
+{
 
-void WorkerWrapper::onFrame() {
     BWAPI::Unit closest = nullptr;
     int smallestDistance = INT32_MAX;
-    if (this->currJob == Jobs::None) {
-        for (const auto& i : BWAPI::Broodwar->getMinerals()) {
-            if (this->unit->getDistance(i) < smallestDistance) {
+
+    // if not busy or working
+    if (this->currJob == Jobs::None || !this->isBusy())
+    { // get command from strategist
+        auto order = Strategist::getInstance().getUnitOrder(this->type);
+
+        if (order.has_value()) // if build order recieved
+        {                      // store results
+            this->buildOrder = order.value();
+            this->currJob = Jobs::MorphIntoNewBuildings;
+        }
+    }
+
+    // not busy + build order
+    if (this->buildOrder != BWAPI::UnitTypes::Unknown && !this->isBusy())
+    { // get closest bweb block to the starting location of the right size
+        auto tiles = BWEB::Blocks::getClosestBlock(BWAPI::Broodwar->self()->getStartLocation())->getPlacements(this->buildOrder);
```

Comment:
There are going to be issues with this once we get the strategist to make decisions. Since you're only grabbing the nearest block to the starting point, it will grab the same block every time. We can test this using a build order that creates multiple buildings, like:
```
    {BWAPI::UnitTypes::Zerg_Drone, 8},
    {BWAPI::UnitTypes::Zerg_Spawning_Pool, 10},
    {BWAPI::UnitTypes::Zerg_Drone, 8},
    {BWAPI::UnitTypes::Zerg_Drone, 10},
    {BWAPI::UnitTypes::Zerg_Zergling, 12},
    {BWAPI::UnitTypes::Zerg_Zergling, 14},
    {BWAPI::UnitTypes::Zerg_Zergling, 16},
    {BWAPI::UnitTypes::Zerg_Overlord, 18},
    {BWAPI::UnitTypes::Zerg_Drone, 18},
    {BWAPI::UnitTypes::Zerg_Drone, 20},
    {BWAPI::UnitTypes::Zerg_Drone, 22},
    {BWAPI::UnitTypes::Zerg_Hatchery, 24},
    {BWAPI::UnitTypes::Zerg_Hatchery, 22}
    ```